cli/pkg/kctrl/cmd/package/builder/create.go

```diff
@@ -0,0 +1,450 @@
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/package/builder/common"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/package/builder/fetch"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/package/builder/template"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/logger"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/util"
+	kappctrlapis "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apiserver/apis/datapackaging/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+)
+
+type CreateOptions struct {
+	ui          ui.UI
+	depsFactory cmdcore.DepsFactory
+	logger      logger.Logger
+
+	NamespaceFlags cmdcore.NamespaceFlags
+	Name           string
+
+	ValuesSchema      bool
+	DefaultValuesFile string
+
+	pkgCmdTreeOpts cmdcore.PackageCommandTreeOpts
+}
+
+func NewCreateOptions(ui ui.UI, logger logger.Logger, pkgCmdTreeOpts cmdcore.PackageCommandTreeOpts) *CreateOptions {
+	return &CreateOptions{ui: ui, logger: logger, pkgCmdTreeOpts: pkgCmdTreeOpts}
+}
+
+func NewCreateCmd(o *CreateOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "create",
+		Aliases: []string{"g"},
+		Short:   "Create a package",
+		Args:    cobra.ExactArgs(1),
+		RunE:    func(_ *cobra.Command, args []string) error { return o.Run(args) },
+		Example: cmdcore.Examples{
+			cmdcore.Example{"Create a package",
+				[]string{"package", "builder", "create"},
+			},
+		}.Description("", o.pkgCmdTreeOpts),
+		SilenceUsage: true,
+		Annotations:  map[string]string{"table": ""},
+	}
+
+	return cmd
+}
+
+func (o *CreateOptions) Run(args []string) error {
+	//TODO Rohit Should we provide an option to give pkg location?
+	pkgLocation := GetPkgLocation()
+	createStep := NewCreateStep(o.ui, pkgLocation)
+	err := createStep.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type CreateStep struct {
+	Ui           ui.UI
+	PkgLocation  string
+	PkgVersion   string
+	FqName       string
+	ValuesSchema v1alpha1.ValuesSchema
+	Fetch        fetch.FetchStep
+	Template     template.TemplateStep
+	Maintainers  []v1alpha1.Maintainer
+}
+
+func NewCreateStep(ui ui.UI, pkgLocation string) *CreateStep {
+	return &CreateStep{
+		Ui:          ui,
+		PkgLocation: pkgLocation,
+	}
+}
+
+func (createStep *CreateStep) Run() error {
+	err := createStep.PreInteract()
+	if err != nil {
+		return err
+	}
+	err = createStep.Interact()
+	if err != nil {
+		return err
+	}
+	err = createStep.PostInteract()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (createStep CreateStep) getStartBlock() []byte {
+	str := fmt.Sprintf(`
+# Lets start on the package creation process.
+# First we need a directory to store all configurations
+# Creating directory %s
+# 	$ mkdir -p %s
```

Comment:
I feel that this is a bit verbose. Maybe we can reword the texts in such a way that simply mention what we are doing instead of saying something like `Let's do this`, and we can use `#` when we actually need to explain something complex?
```suggestion
	str := fmt.Sprintf(`Starting package creation process...
Creating directory %s to store all configurations
$ mkdir -p %s
```

Reply:
Maybe we can use the [statusLoggingUI](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/a658317283e7a59a0990c1ce0c36729e9d205bf3/cli/pkg/kctrl/cmd/core/status_logging_ui.go#L17)  to show the actual commands that we would run.

Reply:
Same for other places.

Reply:
Yup, we have to write some kind of wrapper on UI. Will need to work on that. Basically there will be two UI's implementation. one for interactive and another for non interactive. Havent picked this item yet.